libopeniscsiusr/Makefile

```diff
@@ -0,0 +1,100 @@
+# Makefile
+#
+# Copyright (C) 2017 Red Hat, Inc.
+# Gris Ge <fge@redhat.com>
+#
+
+ifeq ($(TOPDIR),)
+	TOPDIR = ..
+endif
+
+DESTDIR ?=
+prefix ?= /usr
+INSTALL ?= install
+
+ifndef LIB_DIR
+	ifeq ($(shell test -d /lib64 && echo 1),1)
+		LIB_DIR=$(prefix)/lib64
+	else
+		LIB_DIR=$(prefix)/lib
+	endif
+endif
+
+INCLUDE_DIR ?= $(prefix)/include
+PKGCONF_DIR ?= $(LIB_DIR)/pkgconfig
+
+LIBISCSI_USR_DIR=$(TOPDIR)/libopeniscsiusr
+
+LIBISCSI_USR_VERSION_MAJOR=0
+LIBISCSI_USR_VERSION=0.1.0
+SONAME=$(LIBISCSI_USR_VERSION)
+DEVLIB = libopeniscsiusr.so
+LIBS = $(DEVLIB).$(SONAME)
+LIBS_MAJOR = $(DEVLIB).$(LIBISCSI_USR_VERSION_MAJOR)
+PKGFILE = libopeniscsiusr.pc
+HEADERS = libopeniscsiusr/libopeniscsiusr.h \
+	  libopeniscsiusr/libopeniscsiusr_common.h
+TESTS = tests/test_context
+EXTRA_MAN_FILES = libopeniscsiusr.h.3
+
+OBJS = context.o misc.o
+
+CFLAGS ?= -O2 -g
+CFLAGS += -Wall -Werror -Wextra -fvisibility=hidden -fPIC
+
+LIBADD =
+
+all: $(LIBS) $(LIBS_MAJOR) $(TESTS) doc
+
+$(LIBS): $(OBJS)
+	$(CC) $(LDFLAGS) -shared -Wl,-soname=$@ -o $@ $(OBJS) $(LIBADD)
+	ln -sf $@ $(DEVLIB)
+
+$(LIBS_MAJOR):
+	ln -sf $(LIBS) $@
+
+clean:
+	$(RM) vgcore* core *.a *.o *.gz *.so *.so.* $(TESTS)
+	$(RM) -r docs/man
+
+$(TESTS): CFLAGS = -Wall -Werror -Wextra -I$(TOPDIR)/libopeniscsiusr -g
+$(TESTS): LDFLAGS += $(LIBADD) -L$(TOPDIR)/libopeniscsiusr -lopeniscsiusr
+$(TESTS): $(LIBS)
+
+check: $(LIBS) $(TESTS)
+	sudo env LD_LIBRARY_PATH=$(LIBISCSI_USR_DIR) TESTS="$(TESTS)" \
+		tests/runtest.sh || exit 1;
+
+install: $(LIBS) $(LIBS_MAJOR)
+	$(INSTALL) -d $(DESTDIR)/$(LIB_DIR)/
+	$(INSTALL) -d $(DESTDIR)/$(INCLUDE_DIR)/
+	$(INSTALL) $(LIBS) $(DESTDIR)$(LIB_DIR)/
+	$(INSTALL) $(DEVLIB) $(DESTDIR)$(LIB_DIR)/
+	$(INSTALL) $(LIBS_MAJOR) $(DESTDIR)$(LIB_DIR)/
+	$(INSTALL) $(HEADERS) $(DESTDIR)$(INCLUDE_DIR)/
+	$(INSTALL) $(PKGFILE) $(DESTDIR)$(PKGCONF_DIR)/
```

Comment:
This line actually fails in my testing, since there is no $(PKGFILE) (which is libopeniscsiusr.pc, which does not exist in the git repository). I commented it out (removed it) and the install worked correctly

Reply:
The line I mean is the "$(INSTALL) $(PKGFILE) ..." line.